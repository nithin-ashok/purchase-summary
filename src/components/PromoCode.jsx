import React from "react";

class PromoCode extends React.Component {
    constructor(props) {
        super(props);
        this.state={
            value: ""
        }

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.getLabel = this.getLabel.bind(this);
    }

    handleChange(e) {
        e.preventDefault();
        this.setState({
            value: e.target.value
        });
    }

    handleSubmit(e) {
        const {validatePromoCode} = this.props;
        e.preventDefault();
        validatePromoCode(this.state.value);
    }

    getLabel() {
        const {promoApplied, invalidPromo} = this.props;
        if(invalidPromo) {
            return (<span className="font-red"> Invalid Promo</span>)
        }

        if(promoApplied) {
            return (<span className="font-red">Promo Already Applied</span>)
        }

        return "Promo Code";

    }

    render() {
        const {label, loading} = this.props;
        const submitLabel = loading === "PROMO_CODE" ? "Applying..." : "Apply";
        return(
            <form onSubmit={this.handleSubmit} className="form-padding">
                <label htmlFor="promo" className="font-darkgrey font-bold font-small">
                    {this.getLabel()}
                </label>
                <br />
                <input id="promo" className="form-input-text" onChange={this.handleChange} value={this.state.value} />
                <input className="form-input-submit" type="submit" value={submitLabel} />
            </form>
        )
    }

}

export default PromoCode;