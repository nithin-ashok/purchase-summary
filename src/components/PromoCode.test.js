import React from 'react';
import PromoCode from "./PromoCode";
import renderer from 'react-test-renderer';

it("renderes correctly PromoCode component", () => {
    const tree = renderer.
    create(<PromoCode />).toJSON();

    expect(tree).toMatchSnapshot();
});
