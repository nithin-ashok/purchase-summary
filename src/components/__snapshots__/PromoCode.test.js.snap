// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes correctly PromoCode component 1`] = `
<form
  className="form-padding"
  onSubmit={[Function]}
>
  <label
    className="font-darkgrey font-bold font-small"
    htmlFor="promo"
  >
    Promo Code
  </label>
  <br />
  <input
    className="form-input-text"
    id="promo"
    onChange={[Function]}
    value=""
  />
  <input
    className="form-input-submit"
    type="submit"
    value="Apply"
  />
</form>
`;
